Cake.DocumentDb/Requests/DocumentOperations.cs

```diff
@@ -196,28 +197,36 @@ internal async Task PerformTask(IMigrationTask task, Action<JObject> mapAction)
                 task.DatabaseName,
                 task.CollectionName);
 
-            var queryable = client.CreateDocumentQuery<JObject>(collectionResource.SelfLink,
-                new FeedOptions { EnableCrossPartitionQuery = true });
+            var documentQuery = client
+                .CreateDocumentQuery<JObject>(
+                    collectionResource.SelfLink,
+                    new FeedOptions { EnableCrossPartitionQuery = true })
+                .AsDocumentQuery();
 
-            var documentQuery = queryable.AsDocumentQuery();
             var isMatch = task.Filter?.Compile() ?? (doc => true);
 
             while (documentQuery.HasMoreResults)
             {
                 var documents = await documentQuery.ExecuteNextAsync<JObject>();
 
+                var upsertTasks = new List<Task>();

```

Comment:
I'm a bit nervous that 100 tasks running at once may be a bit taxing for a release instance

Reply:
As discussed, starting 100 tasks should not overwhelm the machine as the threadpool and work queue will limit the number of threads and queue tasks until threads become available, see the managed thread pool [docs](https://docs.microsoft.com/en-us/dotnet/standard/threading/the-managed-thread-pool).

I have however added logic to limit number of parallel tasks based on current throughput to reduce the chances of overwhelming Cosmos.